src/IdentityServer4.RavenDB.Storage/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -11,17 +12,21 @@ internal static class ServiceCollectionExtensions
         public static IServiceCollection AddConfigurationDocumentStoreHolder(this IServiceCollection services, Action<RavenDbConfigurationStoreOptions> configureStoreOptions)
         {
             var options = RavenDbStoreOptionsHelper.GetOptions(configureStoreOptions);
-            
-            services.AddSingleton(provider => new ConfigurationDocumentStoreHolder(options));
+
+            services.AddSingleton(provider => options.ResolveDocumentStoreFromServices
+                ? new ConfigurationDocumentStoreHolder(provider.GetRequiredService<IDocumentStore>(), options)

```

Comment:
There are a few issues with this approach. 
Firstly, it introduces redundant complexity, which in turn requires additional validation. What if `ResolveDocumentStoreFromServices` is true and the `IDocumentStore` has not been added to the services container?
Secondly, a user might want to provide the library with their own `IDocumentStore` which is not necessarily resolved by the DI mechanism directly. For example, it might be a property existing on some other object.  This is the pattern that we often apply in our internal projects at HibernatingRhinos. In that case we would need another option pointing to where the library should look for the `IDocumentStore` object.

What about we add at least two more `IIdentityServerBuilder` extension methods? One of them accepting `IDocumentStore` object and another one taking a `Func<T>`  provider as a parameter.  That way we would let users inject their DocumentStores wherever they want from.